doc: update os.tmpdir() to discuss symlink results on macOS (#31877)
nodejs/node: doc/api/os.md

Returns the operating system's default directory for temporary files as a
string.

> On macOS, the returned path is usually a symlink. To get the resolved path, use

Reviewer: usually? Can we document the conditions where it either is or isn't?

Author: > Can we document the conditions where it either is or isn't?

I'll replace "usually" with "by default".

Author: ```suggestion
By default on macOS, the returned path is a symlink. To get the resolved path, use
```

[`fs.realpath()`][] or [`fs.realpathSync()`][].

```js
const tmpdirPath = os.tmpdir();

// For most use cases, it does not matter whether tmpdirPath is a symlink or not

> // but if you need it to be a resolved path, use fs.realpath() or

Reviewer: ```suggestion
// but if a resolved path is needed , use fs.realpath() or
```

Author: Blech, passive voice, gross, I'm going with this instead:

```suggestion
// but to guarantee a resolved path, use fs.realpath() or
```
